compass/landice/tests/mesh_modifications/subdomain_extractor/extract_region.py

```diff
@@ -0,0 +1,176 @@
+            # some funky region boundaries near ice-shelf fronts.
+            noRegionMask = (np.squeeze(regionCellMasks.sum(axis=1)) == 0)
+            thickness = dsMesh['thickness'][:].values
+            bed = dsMesh['bedTopography'][:].values
+            oceanMask = np.squeeze((thickness[0, :] == 0.0) * (bed[0, :] <=
+                                                               0.0))
+            floatMask = np.squeeze(((thickness[0, :] * 910.0 / 1028.0 +
+                                     bed[0, :]) < 0.0) *
+                                   (thickness[0, :] > 0))
+            growMask = np.logical_or(noRegionMask, oceanMask)
+            print(f'sum norregion={growMask.sum()}, {growMask.shape}')
+            conc = dsMesh['cellsOnCell'][:].values
+            neonc = dsMesh['nEdgesOnCell'][:].values
+
+            # First grow forward to capture any adjacent ice shelf
+            print('Starting floating ice fill')
+            iter = 0
+            nMaskCells = keepMask.sum()
+            while True:
+                maskInd = np.nonzero(keepMask == 1)[0]
+                print(f'iter={iter}, keepMask size={keepMask.sum()}')
+                newKeepMask = keepMask.copy()
+                for iCell in maskInd:
+                    neighs = conc[iCell, :neonc[iCell]] - 1
+                    neighs = neighs[neighs >= 0]  # drop garbage cell
+                    for jCell in neighs:
+                        if floatMask[jCell] == 1:
+                            newKeepMask[jCell] = 1
+                keepMask = newKeepMask.copy()
+                nMaskCellsNew = keepMask.sum()
+                if nMaskCellsNew == nMaskCells:
+                    break
+                nMaskCells = nMaskCellsNew
+                iter += 1
+
+            print('Starting ocean grow fill')
+            # find cells at current ocean bdy
+            for iter in range(grow_iters):
+                maskInd = np.nonzero(keepMask == 1)[0]
+                print(f'iter={iter}, keepMask size={keepMask.sum()}')
+                newKeepMask = keepMask.copy()
+                for iCell in maskInd:
+                    neighs = conc[iCell, :neonc[iCell]] - 1
+                    neighs = neighs[neighs >= 0]  # drop garbage cell
+                    for jCell in neighs:
+                        if growMask[jCell] == 1:
+                            newKeepMask[jCell] = 1
+                keepMask = newKeepMask.copy()
+
+        # To call 'cull' with an inverse mask, we need a dataset with the
+        # mask saved to the field regionCellMasks
+        outdata = {'regionCellMasks': (('nCells', 'nRegions'),
+                                       keepMask.reshape(nCells, 1))}
+        dsMaskOut = xarray.Dataset(data_vars=outdata)
+        # For troubleshooting, one may want to inspect the mask, so write out
+        # (otherwise not necessary to save to disk)
+        write_netcdf(dsMaskOut, 'cull_mask.nc')
+
+        # cull the mesh
```

Comment:
Note that I'm planning to refactor the mesh test cases soon, which will probably move a lot of these mesh operations (down to ~L158) to `compass.landice.mesh` to reduce repetition between test cases.